CRM/Civicase/Form/Report/BaseExtendedReport.php

```diff
@@ -753,6 +765,22 @@ public function getContactColumnOptions(array $spec, array $usedIds) {
     return $options;
   }
 
+  /**
+   * Restrict rows to only used options in where clause.
+   */
+  public function extendedWhere() {
```

Comment:
Don't think we should implement this for all fields for rows except the contact reference fields. The requirement was to remove columns without results but it might lead to unexpected results if there are some rows cut off because they were removed.

What do you think?